packages/plugin-ext-metrics/src/browser/plugin-metrics-output-registry.ts

```diff
@@ -0,0 +1,64 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { OutputChannelRegistryMainImpl } from '@theia/plugin-ext/lib/main/browser/output-channel-registry-main';
+import { PluginMetricsCreator } from './plugin-metrics-creator';
+import { createDefaultRequestData } from './plugin-metrics-interfaces';
+
+@injectable()
+export class PluginMetricsOutputChannelRegistry extends OutputChannelRegistryMainImpl {
+
+    @inject(PluginMetricsCreator)
+    protected readonly pluginMetricsCreator: PluginMetricsCreator;
+
+    // This is a map of output channel names to plugin ids
+    private supportedExtOutputChannelToID = new Map<string, string>();
+
+    constructor() {
+        super();
+        this.supportedExtOutputChannelToID.set('YAML Support', 'redhat.vscode-yaml');
```

Comment:
Should not we pass `PluginInfo` to `$append` method as well?

Reply:
I've created a PR: https://github.com/eclipse-theia/theia/pull/6312 once thats merged i'll update this